File: Data Structures & Algorithms/binary-tree-preorder-traversal/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def preorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        
        ans =[]
        self.traverse(ans,root)
        return ans
    
    def traverse(self,ans,node):
        if node is None:
            return
        
        ans.append(node.val)
        self.traverse(ans,node.left)
        self.traverse(ans,node.right)

File: Data Structures & Algorithms/binary-tree-inorder-traversal/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> inorderTraversal(TreeNode root) {
        // if (root == null){
        //     return null;
        // }

        List<Integer> nodes = new ArrayList<>();
        traverse(nodes,root);
        return nodes;
        
    }
    private void traverse(List<Integer> nodes, TreeNode root){

        if (root == null){
            return ;
        }
        traverse(nodes,root.left);
        nodes.add(root.val);
        traverse(nodes,root.right);
    }
}